launch/launcher_test.go

```diff
@@ -229,13 +239,126 @@ func testLauncher(t *testing.T, when spec.G, it spec.S) {
+							filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "exec.d"),
+							filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "exec.d", "start"),
+						)
+						mkfile(t, "",
+							filepath.Join(tmpDir, "launch", "0.4_buildpack", "layer4", "exec.d", "exec_d_1"),
+							filepath.Join(tmpDir, "launch", "0.4_buildpack", "layer4", "exec.d", "start", "exec_d_1"),
+							filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "exec.d", "exec_d_1"),
+							filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "exec.d", "exec_d_2"),
+							filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "exec.d", "start", "exec_d_1"),
+							filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "exec.d", "start", "exec_d_2"),
+						)
+					})
+
+					it("should run exec.d binaries after static env files", func() {
+						gomock.InOrder(
+							env.EXPECT().AddRootDir(filepath.Join(tmpDir, "launch", "0.3_buildpack", "layer1")),
+							env.EXPECT().AddRootDir(filepath.Join(tmpDir, "launch", "0.3_buildpack", "layer2")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.3_buildpack", "layer1", "env")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.3_buildpack", "layer1", "env.launch")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.3_buildpack", "layer2", "env")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.3_buildpack", "layer2", "env.launch")),
+
+							env.EXPECT().AddRootDir(filepath.Join(tmpDir, "launch", "0.4_buildpack", "layer3")),
+							env.EXPECT().AddRootDir(filepath.Join(tmpDir, "launch", "0.4_buildpack", "layer4")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.4_buildpack", "layer3", "env")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.4_buildpack", "layer3", "env.launch")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.4_buildpack", "layer4", "env")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.4_buildpack", "layer4", "env.launch")),
+
+							env.EXPECT().AddRootDir(filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "env")),
+							env.EXPECT().AddEnvDir(filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "env.launch")),
+
+							execd.EXPECT().ExecD(
+								filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "exec.d", "exec_d_1"),
+								env,
+							),
+							execd.EXPECT().ExecD(
+								filepath.Join(tmpDir, "launch", "0.5_buildpack", "layer5", "exec.d", "exec_d_2"),
+								env,
+							),
+						)
+						h.AssertNil(t, launcher.LaunchProcess("", process))
+						if len(syscallExecArgsColl) != 1 {
+							t.Fatalf("expected syscall.Exec to be called once: actual %v\n", syscallExecArgsColl)
+						}
+					})
+
+					when("process is buildpack-provided", func() {
+						it.Before(func() {
+							process.Type = "start"
```

Comment:
In reading through these tests at first I thought "start" had some special meaning - maybe it's clearer to call it `some-process`